### DISCUSSION TITLE
How to understand the role of top-level signals in Rocket?

### DISCUSSION THREAD

**Discussion by tengwu** [chipsalliance/rocket-chip#3381] [Q&A] [answered]
**Type of issue**: bug report

**Impact**: unknown

**Development Phase**: request

**Other information**

Recently, I've been trying to run Rocket Core on the ZCU102 development board. I've previously attempted the [rc-fpga-zcu](https://github.com/li3tuo4/rc-fpga-zcu) project and the [labeled-RISC-V](https://github.com/LvNA-system/labeled-RISC-V/tree/master/fpga) project, both of which were able to run specific versions of Rocket Core on the ZCU102 board. However, it has been two years since then, and the current version of Rocket Core's top-level module has many additional signals. I'm wondering where I can quickly find information about the purpose of these signals, so that I can efficiently map Rocket Core to my FPGA.

Any guidance or resources on understanding the functionality of these new signals would be greatly appreciated.

Thank you!

**Comment by ZenithalHourlyRate** [comment, marked as answer]
Hi, you can list these signals and I can offer some explanation.

A general guide is to `grep` part of the signal name and find the corresponding scala file.

**Reply by tengwu** [reply]
Thanks for your advice and I will have a try.
